### DISCUSSION TITLE
Using DeePMD potentials with hybrid atom_style

### DISCUSSION THREAD

**Discussion by kuntalg97** [deepmodeling/deepmd-kit#3942] [General]
Hi,

I'm trying to set up a system where a subset of atoms of the whole system is to be treated with DeePMD potentials, and the remainder of the system with classical force fields. Now, I need to define bonds and angles for the non-DeePMD part of the system.

For this, I'm using hybrid atom_style atomic angle

Here is a part of my LAMMPS data file :

1506 atoms
1000 bonds
500 angles
0 dihedrals
0 impropers
5 atom types
1 bond types
1 angle types
0 dihedral types
0 improper types
0.000000 41.500000  xlo xhi
0.000000 41.500000  ylo yhi
0.000000 41.500000  zlo zhi

Atoms    # atom-id atom-type x y z mol-id

1   2       15.867987       11.342518       33.834008   1
2   1       16.436384       10.828119       37.357549   1
.
.
.

(Bonds and angles defined later, and masses defined in the LAMMPS input script. Charges also defined using fix property/atom)

This is my pair style : pair_style      hybrid/overlay deepmd graph.pb lj/charmm/coul/long 8 10

With this, and when I run NVT, I get a segmentation fault as error, which I'm not able to fix. If I turn off the deepmd part and replace it with simple LJ interactions, the simulations run fine.

Can pair_style deepmd not handle hybrid atom_style interactions?

Could you please help me fix this? I realize it's a super specific problem, but any help would be appreciated.

**Comment by njzjz** [comment]
I assume you mean "hybrid pair_style". Do you use the latest version? There was a bug #1649 several years ago.

**Reply by kuntalg97** [reply]
No, I'm referring to hybrid atom_style. I'm combining atom_styles atomic (which DeePMD requires) and angle (which allows bond and angle definitions).

I'm using LAMMPS (2 Aug 2023) version, DeePMD version 2.2.9 and tensorflow version 2.15.0.

**Reply by njzjz** [reply]
DeePMD-kit doesn't require `atomic` - there is the unit test for `full`.

https://github.com/deepmodeling/deepmd-kit/blob/60927373ded5d61c4c9181e42f8f05cd0314cb04/source/lmp/tests/test_dplr.py#L300

**Reply by kuntalg97** [reply]
Okay, so I used atom_style full, but I still ended up getting the same error (segmentation fault).
